table/block_based/filter_policy.cc

```diff
@@ -209,10 +224,36 @@ class FastLocalBloomBitsBuilder : public XXPH3FilterBitsBuilder {
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) override {
     size_t num_entries = hash_entries_.size();
     size_t len_with_metadata = CalculateSpace(num_entries);
+    const size_t bytes_hash_entries =
+        (num_entries == 0 ? 0 : num_entries * sizeof(hash_entries_[0]));
+
+    // Cache reservation for hash entries
+    std::unique_ptr<CacheReservationHandle<CacheEntryRole::kFilterConstruction>>
```

Comment:
Discussed offline: 
(1) try charge memory of a bucket of keys instead of per key. sizeof(bucket) = dummy entry size/sizeof (key hash)
(2) check the performance